client/my-sites/themes/themes-banner/index.jsx

```diff
@@ -3,34 +3,105 @@
 /**
  * External dependencies
  */
+
 import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+
+import { localize } from 'i18n-calypso';
 import PropTypes from 'prop-types';
-import cssSafeUrl from 'lib/css-safe-url';
 import Button from 'components/button';
+import safeImageUrl from 'lib/safe-image-url';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { recordTracksEvent } from 'state/analytics/actions';
+import { getActiveTheme, getThemeDetailsUrl } from 'state/themes/selectors';
 
-export class ThemesBanner extends PureComponent {
+class ThemesBanner extends PureComponent {
 	static propTypes = {
-		title: PropTypes.string.isRequired,
-		description: PropTypes.string.isRequired,
-		action: PropTypes.func,
-		actionLabel: PropTypes.string.isRequired,
-		backgroundImage: PropTypes.string,
-		href: PropTypes.string,
+		themeId: PropTypes.string,
+		themeName: PropTypes.string,
+		title: PropTypes.node.isRequired,
+		description: PropTypes.node.isRequired,
+		buttonLabel: PropTypes.string.isRequired,
+		backgroundColor: PropTypes.string,
+		image: PropTypes.string,
+		imageAttrs: PropTypes.shape( {
+			width: PropTypes.number,
+		} ),
+		imageTransform: PropTypes.string,
+
+		// Connected props
+		siteId: PropTypes.number,
+		themeUrl: PropTypes.string,
+		activeThemeId: PropTypes.string,
+	};
+
+	onClick = () => {
+		const { siteId, themeId, activeThemeId } = this.props;
+		const tracksData = {
+			site_id: siteId,
+			theme: themeId,
+			active_theme: activeThemeId,
+		};
+		recordTracksEvent( 'calypso_showcase_banner_click', tracksData );
```

Comment:
👍

Reply:
Oops, I missed that we were directly _calling_ the `recordTracksEvent`  action here. That won't work; instead, we have to `dispatch()` it. The preferred way to do that in a React component is to use `connect()`'s second argument, [`mapDispatchToProps`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options), and use the "object shorthand" there.

I'll follow up with a code sample...

Reply:
```diff
diff --git a/client/my-sites/themes/themes-banner/index.jsx b/client/my-sites/themes/themes-banner/index.jsx
index 072be31..bfafcdb 100644
--- a/client/my-sites/themes/themes-banner/index.jsx
+++ b/client/my-sites/themes/themes-banner/index.jsx
@@ -16,7 +16,7 @@ import PropTypes from 'prop-types';
 import Button from 'components/button';
 import safeImageUrl from 'lib/safe-image-url';
 import { getSelectedSiteId } from 'state/ui/selectors';
-import { recordTracksEvent } from 'state/analytics/actions';
+import { recordTracksEvent as recordTracksEventAction } from 'state/analytics/actions';
 import { getActiveTheme, getThemeDetailsUrl } from 'state/themes/selectors';
 
 class ThemesBanner extends PureComponent {
@@ -37,7 +37,7 @@ class ThemesBanner extends PureComponent {
        };
 
        recordEvent = () => {
-               const { siteId, themeId, activeThemeId } = this.props;
+               const { siteId, themeId, activeThemeId, recordTracksEvent } = this.props;
                const tracksData = {
                        site_id: siteId,
                        theme: themeId,
@@ -99,4 +99,6 @@ const mapStateToProps = ( state, { themeId } ) => {
        };
 };
 
-export default connect( mapStateToProps )( localize( ThemesBanner ) );
+export default connect( mapStateToProps, {
+       recordTracksEvent: recordTracksEventAction,
+} )( localize( ThemesBanner ) );
```

(untested)

Reply:
So this essentially just adds an object as `connect`'s second argument, whose values will be bound to `dispatch()` by `connect()`. The `dispatch()`-bound actions will then be passed on to the wrapped component as props (using the object keys as prop names).

Reply:
The rest of the diff essentially makes sure that the `dispatch()`-bound `recordTracksEvent` action prop name doesn't shadow the unbound one that we import from `state/analytics/actions`.